lib/request.js

```diff
@@ -209,23 +212,28 @@ Object.defineProperties(Request.prototype, {
   },
   host: {
     get () {
-      return this.raw.headers.host || this.raw.headers[':authority']
+      let host = this.headers.host ?? this.headers[':authority']
+      // support http.requireHostHeader === false
+      if (this.server.server.requireHostHeader === false) host ??= ''
+      return host
     }
   },
   hostname: {
     get () {
-      return (this.host).split(':')[0]
+      return this.host.split(':')[0]
     }
   },
   port: {
     get () {
       // first try taking port from host
-      const portFromHost = parseInt((this.host).split(':').slice(-1)[0])
+      const portFromHost = parseInt(this.host.split(':').slice(-1)[0])
       if (!isNaN(portFromHost)) {
         return portFromHost
       }
       // now fall back to port from host/:authority header
-      const portFromHeader = parseInt((this.headers.host || this.headers[':authority']).split(':').slice(-1)[0])
+      let host = (this.headers.host ?? this.headers[':authority'])
+      if (this.server.server.requireHostHeader === false) host ??= ''
```

Comment:
Using the `http.Server` for checking the properties, it is better because it may exists in `http`, `https` or `http2` options. It should be always the same when we use multiple server instance.